### DISCUSSION TITLE
Staging updated and new files together -updates the new file contents in Git extension

### DISCUSSION THREAD

**Discussion by abeed100** [gitextensions/gitextensions#9847] [Q&A] [answered]
I have a Visual Studio Repository under Git. I opened it in Git extension. Now added few new files and updated few files  in a folder under the project(VS2017-->Add files-->exiting files..). Saved All (In Vs). Now I saw list of files in Gi extensions pane (Unstaged). They look fine. But when I select all of the files and Stage all together, the new files are updated with existing old files and shows up in  the Staged pane/screen. Example:
Unstaged:
-file1_101_10 (This file was updated)
-file2_101_11 (This file was updated)
+ filenew_105_15 (This is the new file that was added)

After selecting all 3 files clicking Stage All icon
Staged:
-file1_101_10 (remains same)
-file2_101_11 (remains same)
--filenew_105_15 (shows updated with the contents of file2-101_11)

why this happens? After Staged, Why I can't see the 3rd file just a new file with its own contents and nothing from file2)
I can't commit this change as the 3rd file contents changed.

if I do 2 commits--1 for the updated and another for the new, everything work perfect. The new 3rd file remains as it is. But why I can;t do it in 1 commit. Please let me know. Help much needed.

**Comment by mstv** [comment, marked as answer]
Do your filenames start with hyphen(s)? Then they might be misinterpreted as option arguments if GE does not quote them correctly.
You could have a look at GE's Git Command Log (`F12`) and share the essential git command lines and try to run them manually.

**Reply by abeed100** [reply]
No, they don't start with hyphen(-). These are actually:
file1_101_10 (This file was updated)
file2_101_11 (This file was updated)
filenew_105_15 (This is the new file that was added)

**Comment by RussKie** [comment]
Please provide an animation or at least screenshots.

**Reply by abeed100** [reply]
![new-f1](https://user-images.githubusercontent.com/52586569/152003999-4a219ead-f415-4e4b-81aa-4b24ab277639.jpg)

Hi: Here is a copy of it after Staged.

**Reply by gerhardol** [reply]
F12 Git commands is needed (you can edit the log, no need for screenshot)

**Reply by abeed100** [reply]
What is that--F12 Git command. How do I do it from Git extension?  I am in Git extension and the image shows the current state of the Staged files--both updated and added ones. 147902_3141_en.html is a newly added file but it is being updated with the other (updated) file-126801_1130_en.html. Thanks

**Reply by pmiossec** [reply]
It would have been interesting that you include the icons shown at the left to be sure but it seems to be the normal git behavior (and not a GitExtensions issue) where Git tries to find the file renamed based on the content very similar of the files.

It is just a display "issue" because Git doesn't store any rename information. So maybe it could be misleading in your case but it's not a problem. More information here: https://stackoverflow.com/a/7941544/717372

Here it seems that you are not happy with the result.

Maybe you could try to play with the `merge.renameLimit` git setting (but it could have bad impact at other moment where you want to have this rename detection displayed) but there is little that GitExtensions could do.

But, once again, it's just a display not something wrong under the hood.

**Comment by abeed100** [comment]
Ok--let me commit and see if the files are ok

**Reply by abeed100** [reply]
I saw no overlap happens. It just shows the diff in 2 files in Git extension but when i committed, things are ok.

**Reply by mstv** [reply]
If you are confused by git's rename/copy detection, these options might help:
![image](https://user-images.githubusercontent.com/36601201/152586847-00b628c1-6f82-483a-8021-11ca070cb9cf.png)

**Comment by vbjay** [comment]
Put your mouse over one of these files and read the tooltip.
